syntax = "proto3";

package platformq.graph_intelligence;

// The service for getting insights from the knowledge graph.
service GraphIntelligenceService {
  // Retrieves newly detected communities of users.
  rpc GetCommunityInsights(GetCommunityInsightsRequest) returns (GetCommunityInsightsResponse);
}

// The request message for community insights.
// It could be extended to include parameters like time windows or algorithm types.
message GetCommunityInsightsRequest {
  string tenant_id = 1; // To scope the request to a specific tenant.
}

// A single community of users.
message Community {
  string community_id = 1;
  repeated string user_ids = 2;
  // Could also include things like:
  // string topic = 3; // The central topic of the community
  // float cohesion_score = 4;
}

// The response message containing the list of communities.
message GetCommunityInsightsResponse {
  repeated Community communities = 1;
} 